control-plane/pkg/contract/extensions.go

```diff
@@ -16,7 +16,186 @@
 
 package contract
 
+import v1 "knative.dev/eventing/pkg/apis/eventing/v1"
+
 // IncrementGeneration increments Generation.
 func (x *Contract) IncrementGeneration() {
 	x.Generation++
 }
+
+// NewExactFilter converts the SubscriptionsAPIFilter into the exact dialect of
```

Comment:
would it make sense to place the new experimental filters into its own file

Reply:
yeah the filter implementation here is on the Java side, and I'm going to move them to their own subscriptionsapi package indeed. But this `control-plane/pkg/contract/extensions.go` is not a filters implementation, they're conversion functions/factory functions to convert from the api to the protobuf types, which is why I think their place here makes sense. 

In the code, you'd `import contract` and do `contract.NewAllFilter` to get the correct contract type. This is not filters logic.

Reply:
I saw they work on the api. I was just wondering if they fit better in their own file. similar to the [serdes](https://github.com/knative-sandbox/eventing-kafka-broker/blob/main/control-plane/pkg/contract/serde.go) 

you would be still doing `contract.NewAllFilter`

Reply:
yeah sorry I misread, you said its own file, not package.